kie-server-parent/kie-server-services/kie-server-services-task-assigning/kie-server-services-task-assigning-planning/src/main/java/org/kie/server/services/taskassigning/planning/SolutionSynchronizer.java

```diff
@@ -240,7 +249,14 @@ private TaskAssigningSolution recoverSolution() {
                                                                                                      Suspended),
                                                                                        null,
                                                                                        TaskInputVariablesReadMode.READ_FOR_ALL);
-        context.setLastModificationDate(result.getQueryTime());
+
+        final LocalDateTime nextQueryTime = trimMillis(result.getQueryTime());
+        final LocalDateTime adjustedFirstQueryTime = nextQueryTime != null ? nextQueryTime.minusHours(1) : null;
```

Comment:
SonarCloud complains about potential NPE due to `nextQueryTime` might be null.

Reply:
already fixed